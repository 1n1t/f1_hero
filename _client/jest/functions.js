/* eslint-disable import/prefer-default-export */
export function sel(id) {
	return `[data-test="${id}"]`;
}
