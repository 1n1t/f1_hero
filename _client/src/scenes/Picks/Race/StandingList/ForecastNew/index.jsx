import React from 'react';
import PropTypes from 'prop-types';
import { SortableContainer, SortableElement } from 'react-sortable-hoc';

import Item from '../Item';
import { listItemPropTypes } from '../index';

const ForecastNew = ({ items }) => {
	const SortableItem = SortableElement(item => <Item {...item} />);

	return (
		<div className="standings-container">
			{items.map((value, index) => (
				<SortableItem key={`item-${value.position}`} index={index} {...value} />
			))}
		</div>
	);
};

ForecastNew.propTypes = {
	items: PropTypes.arrayOf(PropTypes.shape(listItemPropTypes)).isRequired
};

export default SortableContainer(ForecastNew);
