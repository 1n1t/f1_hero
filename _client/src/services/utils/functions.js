/* eslint-disable import/prefer-default-export */
export const isObjectEmpty = obj =>
	Object.keys(obj).length === 0 && obj.constructor === Object;

export const getNodePaddings = node => {
	const nodeStyle = window.getComputedStyle(node);
	return (
		parseInt(nodeStyle.getPropertyValue('padding-right'), 10) +
		parseInt(nodeStyle.getPropertyValue('padding-left'), 10)
	);
};
