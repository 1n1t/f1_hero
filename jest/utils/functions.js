const mongoose = require('mongoose');
const transformProps = require('transform-props');

function clearCollection(model) {
	return mongoose.model(model).remove();
}

function clearAllCollections() {
	Promise.all([
		mongoose.model('race').remove(),
		mongoose.model('user').remove(),
		mongoose.model('pick').remove(),
		mongoose.model('driver').remove()
	]).then(() => null);
}

function castToString(arg) {
	return String(arg);
}

function docToObj(docObj) {
	return transformProps(docObj, castToString, '_id');
}

module.exports = {
	clearCollection,
	clearAllCollections,
	docToObj
};
