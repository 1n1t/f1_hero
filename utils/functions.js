function getMaxFiveElements(arr) {
	return arr.sort((a, b) => a < b).slice(0, 5);
}

function capitalizeFirstLetter(string) {
	return string.charAt(0).toUpperCase() + string.slice(1);
}

module.exports = {
	getMaxFiveElements,
	capitalizeFirstLetter
};
