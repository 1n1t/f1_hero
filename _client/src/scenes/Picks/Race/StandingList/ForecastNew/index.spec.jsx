import r from 'Root/jest/utils/renderer';
import pickSample from 'JestClient/samples/pick';
import { sel } from 'JestClient/functions';

import ForecastNew from './index';

const defaultProps = {
	items: pickSample.forecast
};

r.init(ForecastNew, defaultProps);

it('renders correctly', () => {
	const wrapper = r.render(null, true);
	expect(wrapper).toMatchSnapshot();
});

it('renders correct amount of list items', () => {
	const wrapper = r.render(null, true);
	const items = wrapper
		.at(0)
		.dive()
		.find(sel('new-standing-item'));

	expect(items).toHaveLength(defaultProps.items.length);
});
