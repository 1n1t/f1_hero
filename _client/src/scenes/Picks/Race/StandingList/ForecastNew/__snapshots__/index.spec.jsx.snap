// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="standings-container"
>
  <sortableElement
    _driver={
      Object {
        "code": "HAM",
        "name": "L. Hamilton",
      }
    }
    collection={0}
    data-test="new-standing-item"
    index={0}
    key="item-1"
    position={1}
    score={18}
  />
  <sortableElement
    _driver={
      Object {
        "code": "RIC",
        "name": "D. Ricciardo",
      }
    }
    collection={0}
    data-test="new-standing-item"
    index={1}
    key="item-2"
    position={2}
    score={5}
  />
  <sortableElement
    _driver={
      Object {
        "code": "VAN",
        "name": "S. Vandoorne",
      }
    }
    collection={0}
    data-test="new-standing-item"
    index={2}
    key="item-3"
    position={3}
    score={0}
  />
</div>
`;
